### PR TITLE
Add test coverage measuring

### LINKED ISSUE #37: Measure Test Coverage
## Acceptance Criteria

- [ ] When I run the test suite, it tells me how much of my code is being tested.
- [ ] Document how to run the test suite.

## First Step

Figure out how to measure test coverage in Elixir.

### PR DESCRIPTION
**Pull request by tpendragon**
This adds excoveralls (which is the state of the art coverage tool) and a `mix spec` command which will run the whole suite with an HTML export of coverage and fail if the coverage is less than 100%.

I decided on excoveralls based on the HTML view of `mix test --cover` not having a table of contents to see all coverage in one place. Read on https://elixirforum.com/t/code-coverage-tools-for-elixir/18102 that `excoveralls` was the best option, and it seems well maintained.

Closes #37

### CODE REVIEW

**Review by hackartisan — CHANGES_REQUESTED** [review]
Looks good! I tried it, it doesn't seem to print out the coverage when you run `mix test` -- is it determined separately / statically? `mix coveralls` seems to work, but I'm not sure where to see the html view you referenced. Can you add to the readme instructions for how to run / view coverage?

File: mix.exs [hunk 1]
```diff
@@ -9,7 +9,16 @@ defmodule DpulCollections.MixProject do
       elixirc_paths: elixirc_paths(Mix.env()),
       start_permanent: Mix.env() == :prod,
       aliases: aliases(),
-      deps: deps()
+      deps: deps(),
+      test_coverage: [tool: ExCoveralls],
```
Comment (hackartisan) [inline mix.exs:13]: oh i guess this makes `mix test --cover` use excoveralls

**Comment by tpendragon** [comment]
@hackartisan Mix spec or mix coveralls.html should generate the html page.

**Comment by hackartisan** [comment]
@tpendragon can you add that to the readme

**Comment by tpendragon** [comment]
> @tpendragon can you add that to the readme

Yup!

**Review by hackartisan — APPROVED** [review]

**Pull request merged**